x-pack/metricbeat/module/azure/monitor/monitor.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package monitor
+
+import (
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/x-pack/metricbeat/module/azure"
+	"github.com/pkg/errors"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("azure", "monitor", New, mb.DefaultMetricSet())
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	client *Client
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The azure monitor metricset is beta.")
+	var config azure.Config
+	err := base.Module().UnpackConfig(&config)
+	if err != nil {
+		return nil, errors.Wrap(err, "error unpack raw module config using UnpackConfig")
+	}
+	if len(config.Resources) == 0 {
+		return nil, errors.New("no metrics defined")
+	}
+	monitorClient, err := NewClient(config)
```

Comment:
Does this client have to be closed when the metricset is stopped?

Reply:
Not necessary, will need to look into the Context object (context package) though

Reply:
My expectation would be that each metricset is independent. This means each metricset should have it's own client and not be able to affect an other metricset.

Reply:
> My expectation would be that each metricset is independent. This means each metricset should have it's own client and not be able to affect an other metricset.

@ruflin I think I don't follow you, I'd say that as is now each metricset has it's own client.

And I think there is no need to close it because it is only used for HTTP requests and it doesn't keep any resource.

Reply:
If that is the case, I'm ok with it. Reason I brought it up is that in the past we had some leaks when a metricset was stopped and we leaked some handlers.